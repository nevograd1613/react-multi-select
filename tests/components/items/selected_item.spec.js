import React from "react";
import ShallowRenderer from "react-shallow-renderer";

import SelectedItem from "../../../src/components/items/selected_item";

const item = { id: 1, label: "Hi" };

describe("SelectedItem", () => {
  test("default snapshot", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<SelectedItem />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with valid item", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<SelectedItem item={item} />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with group item", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<SelectedItem item={item} group />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with height", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<SelectedItem height={20} />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with disabled item", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <SelectedItem item={{ id: 1, label: "item 1", disabled: true }} />,
    );
    expect(tree).toMatchSnapshot();
  });
});
