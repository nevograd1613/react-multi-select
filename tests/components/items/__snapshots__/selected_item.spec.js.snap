// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedItem default snapshot 1`] = `
<div
  className="selected_item"
  style={
    {
      "height": 40,
    }
  }
>
  <ItemLabel
    label=""
  />
  <WithStyles(IconButton)>
    <MdClose />
  </WithStyles(IconButton)>
</div>
`;

exports[`SelectedItem snapshot with disabled item 1`] = `
<div
  className="selected_item"
  style={
    {
      "height": 40,
    }
  }
>
  <ItemLabel
    label="item 1"
  />
  <WithStyles(IconButton)>
    <MdClose />
  </WithStyles(IconButton)>
</div>
`;

exports[`SelectedItem snapshot with group item 1`] = `
<div
  className="with_grouping"
  style={
    {
      "height": 40,
    }
  }
>
  <ItemLabel
    label="Hi"
  />
</div>
`;

exports[`SelectedItem snapshot with height 1`] = `
<div
  className="selected_item"
  style={
    {
      "height": 20,
    }
  }
>
  <ItemLabel
    label=""
  />
  <WithStyles(IconButton)>
    <MdClose />
  </WithStyles(IconButton)>
</div>
`;

exports[`SelectedItem snapshot with valid item 1`] = `
<div
  className="selected_item"
  style={
    {
      "height": 40,
    }
  }
>
  <ItemLabel
    label="Hi"
  />
  <WithStyles(IconButton)>
    <MdClose />
  </WithStyles(IconButton)>
</div>
`;
